### PR TITLE
[folly,fbthrift,mvfst,fizz,wangle,proxygen]: update to 2023.05.15

### PR DESCRIPTION
**Pull request by jiayuehua**

### CODE REVIEW

**Comment by jiayuehua** [comment]
#28475

File: ports/folly/portfile.cmake [hunk 1]
```diff
@@ -8,16 +8,15 @@ vcpkg_add_to_path("${PYTHON3_DIR}")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebook/folly
-    REF d8ed9cd2869c74b00fa6f1a7603301183f5c2249 #v2022.10.31.00
-    SHA512 55040dadb8a847f0d04c37a2dce920bb456a59decebc90920831998df9671feb33daf1f4235115adcce5eb9c469b97b9d96fa7a67a5914c434ebc1efc04f4770
+    REF v2023.05.15.00 #v2022.10.31.00
```
Comment (Cheney-W) [inline ports/folly/portfile.cmake:11]:
```suggestion
    REF "v${VERSION}"
```

File: ports/proxygen/portfile.cmake [hunk 2]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebook/proxygen
-    REF 0c932f6c57095838494520ec4ce7243a7f0c1234  #v2022.07.11.00
-    SHA512 496189c37be8f42821cb5357e501b02f013858f1917854000777ac5c0403d078837e89dc3eef5f66f30f83c7506e40316d730725d7f97070090ac6549e766093
+    REF v2023.05.15.00
```
Comment (Cheney-W) [inline ports/proxygen/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/wangle/portfile.cmake [hunk 3]
```diff
@@ -3,12 +3,12 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO facebook/wangle
-    REF v2022.03.21.00
-    SHA512 069fe7758ccb2f33665f53109047f2e0158e248fda8c760eddd978afd6980acbc6421db60721bf320deca557369377c8757f13c67f8685e3a98394324e75e72e
+    REF v2023.05.15.00
+    SHA512  aa87614f9630fd325e848bef38c520b85a57eaf0e3dd5cb3421a0bb55322fbdf9c22b153436d703eb30ac9f8c4d02bfb71b9313acd16c412609216eb0ca16109
```
Comment (Cheney-W) [inline ports/wangle/portfile.cmake:7]:
```suggestion
    REF "v${VERSION}"
    SHA512 aa87614f9630fd325e848bef38c520b85a57eaf0e3dd5cb3421a0bb55322fbdf9c22b153436d703eb30ac9f8c4d02bfb71b9313acd16c412609216eb0ca16109
```

File: ports/wangle/portfile.cmake [hunk 4]
```diff
@@ -3,12 +3,12 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
-    REF v2022.03.21.00
-    SHA512 069fe7758ccb2f33665f53109047f2e0158e248fda8c760eddd978afd6980acbc6421db60721bf320deca557369377c8757f13c67f8685e3a98394324e75e72e
+    REF v2023.05.15.00
+    SHA512  aa87614f9630fd325e848bef38c520b85a57eaf0e3dd5cb3421a0bb55322fbdf9c22b153436d703eb30ac9f8c4d02bfb71b9313acd16c412609216eb0ca16109
     HEAD_REF master
     PATCHES
-        fix-config-cmake.patch
-        fix_dependency.patch
+    fix-config-cmake.patch
+    fix_dependency.patch
```
Comment (Cheney-W) [inline ports/wangle/portfile.cmake:11]: Indentation is required here, so no modification is needed.

**Comment by Cheney-W** [comment]
The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded.

**Review by Cheney-W — APPROVED** [review]

File: ports/rsocket/fix-rsockserver-build-error.patch [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+-          std::placeholders::_2));
++      [this, serviceHandler=serviceHandler](
++          std::unique_ptr<DuplexConnection> connection,
++          ResumeParameters resumeParameters) mutable noexcept{
++              this->onRSocketResume(
++                  serviceHandler,
++                  std::move(connection),
++                  resumeParameters
++              );
++        });
```
Comment (dan-shaw) [inline ports/rsocket/fix-rsockserver-build-error.patch:32]: What does this fix?

Comment (jiayuehua) [reply ports/rsocket/fix-rsockserver-build-error.patch:32]: before apply this fix , build error:
[22/59] /usr/bin/c++ -DDEBUG -DGFLAGS_IS_A_DLL=0 -I/home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean -I/home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/yarpl/.. -isystem /home/rob/vcpkg/installed/x64-linux/include -fPIC -Wall -Wextra -Woverloaded-virtual -g -Wno-noexcept-type -Wno-unreachable-code -g -std=gnu++17 -MD -MT CMakeFiles/ReactiveSocket.dir/rsocket/framing/ScheduledFrameTransport.cpp.o -MF CMakeFiles/ReactiveSocket.dir/rsocket/framing/ScheduledFrameTransport.cpp.o.d -o CMakeFiles/ReactiveSocket.dir/rsocket/framing/ScheduledFrameTransport.cpp.o -c /home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/rsocket/framing/ScheduledFrameTransport.cpp
/home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/rsocket/RSocketServer.cpp:138:16: error: cannot convert ‘std::_Bind_helper<fal
se, void (rsocket::RSocketServer::*)(std::shared_ptr<rsocket::RSocketServiceHandler>, std::unique_ptr<rsocket::DuplexConnection>, rsocket::Resume
Parameters) noexcept, rsocket::RSocketServer*, std::shared_ptr<rsocket::RSocketServiceHandler>&, const std::_Placeholder<1>&, const std::_Placeho
lder<2>&>::type’ to ‘rsocket::SetupResumeAcceptor::OnResume’ {aka ‘folly::Function<void(std::unique_ptr<rsocket::DuplexConnection>, rsocket::Resu
meParameters) noexcept>’}
  138 |       std::bind(
      |       ~~~~~~~~~^
      |                |
      |                std::_Bind_helper<false, void (rsocket::RSocketServer::*)(std::shared_ptr<rsocket::RSocketServiceHandler>, std::unique_ptr
<rsocket::DuplexConnection>, rsocket::ResumeParameters) noexcept, rsocket::RSocketServer*, std::shared_ptr<rsocket::RSocketServiceHandler>&, cons
t std::_Placeholder<1>&, const std::_Placeholder<2>&>::type
  139 |           &RSocketServer::onRSocketResume,
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  140 |           this,
      |           ~~~~~ 
  141 |                   serviceHandler,
      |                   ~~~~~~~~~~~~~~~
  142 |           std::placeholders::_1,
      |           ~~~~~~~~~~~~~~~~~~~~~~
  143 |           std::placeholders::_2));
      |           ~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/rsocket/RSocketServer.h:28,
                 from /home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/rsocket/RSocketServer.cpp:15:
/home/rob/vcpkg/buildtrees/rsocket/src/84ec7fc921-09c4d8f7c8.clean/rsocket/internal/SetupResumeAcceptor.h:53:59: note:   initializing argument 3 
of ‘void rsocket::SetupResumeAcceptor::accept(std::unique_ptr<rsocket::DuplexConnection>, rsocket::SetupResumeAcceptor::OnSetup, rsocket::SetupRe
sumeAcceptor::OnResume)’
   53 |   void accept(std::unique_ptr<DuplexConnection>, OnSetup, OnResume);

**Pull request merged**